src/orchard/java.clj

```diff
@@ -223,14 +225,16 @@
 ;; and recompiled, we cache such classes with last-modified property, so that we
 ;; know when to purge those classes from cache.
 
-(def cache (atom {}))
+;; 250 is large enough to be useful (and hold a few key classes, like Object),
+;; and small enough to not incur into OOMs.
+(def ^Map cache (LruMap. 250))
```

Comment:
I'm guessing that even double the size it wouldn't be an issue. You might want to expand the documentation a little bit to elaborate that this cache size is bounded and that's the reason why we went with a custom implementation.